src/engine/test/integration_tests/policy/steps/api_steps.py

```diff
@@ -0,0 +1,185 @@
+from api_communication.client import APIClient
+from google.protobuf.json_format import ParseDict
+from api_communication.proto import policy_pb2 as api_policy
+from api_communication.proto import engine_pb2 as api_engine
+
+from behave import given, when, then
+import os
+import subprocess
+from typing import Optional, Tuple, List
+
+ENV_DIR = os.environ.get("ENV_DIR", "")
+SOCKET_PATH = ENV_DIR + "/queue/sockets/engine-api"
+RULESET_DIR = ENV_DIR + "/engine"
+
+api_client = APIClient(SOCKET_PATH)
+
+def run_command(command):
+    result = subprocess.run(
+        command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    assert result.returncode == 0, f"{result.stderr}"
+
+def send_recv(request, expected_response_type) -> Tuple[Optional[str], dict]:
+    error, response = api_client.send_recv(request)
+    parse_response = ParseDict(response, expected_response_type)
```

Comment:
Check `error`, this var handle the protocol error:

https://github.com/wazuh/wazuh/blob/aeac12615a30969ac9142da37e65ea2368b03435/src/engine/tools/api-communication/src/api_communication/client.py#L137-L139